StardewArchipelago/Locations/CodeInjections/Vanilla/FriendshipInjections.cs

```diff
@@ -395,5 +418,19 @@ private static void SetFriendshipPoints(string npc, double points)
 
             _friendshipPoints[npc] = points;
         }
+
+        private static string ParseInternalName(string name)

```

Comment:
I think I explained a bit in personal conversation but I'll reiterate.

Essentially, there are, and will be, discrepancies between the displayed name of a NPC, and the actual name of a NPC.  Mr. Ginger is the first.  While we could simply update backend to name them their actual name (MisterGinger), this won't solve the issue down the road.  For example, in SVE, three characters are "added": Marlon, Gunther, and Morris.  They aren't new, but their NPC forms are, and to differentiate them from their old counterparts, last names are appended to them.  Marlon becomes MarlonFay, for example.  This gets ugly, quickly.  So it calls for a parse method to get the proper name. 

Granted the method here is pretty old, and could definitely use a better method.  For one, its basically a very primitive dictionary, and so a dictionary could be called instead.  Another could be that we could shift from actual names, to the NPCs display name instead.  Third is we let the user see the actual names of every NPC when they get the heart items for them, which can end up rather poor-looking over time.

Reply:
Everything you just wrote is perfectly valid (and I was aware of... most(?) of it). But my point is that the method itself does not communicate any of that.

It should not swap between them. It should be a move from one name to the other, one-way only. If you need to do it two-way, then a different structure should be used for the other direction. Or, if NPCs are expected to have a truckload of names, there could be a structure that is a list of list of names where they are grouped by the npc they refer to.
At the base, we could have a dictionary of names to translate, instead of this, like I did for the entrance aliases. On a slightly higher level, as I said yesterday, even that, I found ugly and hard to maintain, so there could be a dedicated class for translating names.

So yeah, I don't have a problem with the use case, just with the implementation

Reply:
That's fine.  its primitive given what I know now anyway.  Was just touching base on what you felt would make better sense.